resources/qml/Menus/MaterialMenu.qml

```diff
@@ -70,13 +66,7 @@ Menu
             checkable: true
             checked: model.id == Cura.MachineManager.allActiveMaterialIds[ExtruderManager.extruderIds[extruderIndex]]
             exclusiveGroup: group
-            onTriggered:
-            {
-                var activeExtruderIndex = ExtruderManager.activeExtruderIndex;
-                ExtruderManager.setActiveExtruderIndex(extruderIndex);
-                Cura.MachineManager.setActiveMaterial(model.id);
-                ExtruderManager.setActiveExtruderIndex(activeExtruderIndex);
-            }
```

Comment:
There was a reason this code was in place. See https://github.com/Ultimaker/Cura/pull/2224. It looks like this commit breaks that again.

Reply:
Oh, I will take a look at it. Thank you for the comment

Reply:
I have reverted your workaround and add some comments. The other issues seem to work as well.